### DISCUSSION TITLE
NGINX logging changes

### DISCUSSION THREAD

**Discussion by jfmatth** [piku/piku#338] [Ideas]
Hi everyone,

I noticed that in piku.py, the ```NGINX_COMMON_FRAGMENT``` has the access.log and error.log commented out.

I'd like to turn this on for a few reasons:

1. piku logs now shows all NGINX log entries
2. systemd doesn't capture these by default, so no reason not to have them somewhere

I was trying to find how NGINX is monitored by other systems, and most of them adjust the format of the logs to their liking anyway.

I'm happy to make a PR for this, but thoughts on why not?

**Comment by chr15m** [comment]
@jfmatth I think the best way to enable this is using `UWSGI_INCLUDE_FILE` in the ENV file. You can create `nginx.conf` and add the logging fragment to it to send those logs wherever you want. I personally also get my apps web framework to do their own logging.

**Reply by jfmatth** [reply]
of course!  A brilliant answer @chr15m thank you.

Ill try that.

BTW, am I wrong that systemd doesn't by default pickup nginx logs?  I think it would require a change to journald.conf

**Comment by chr15m** [comment]
Here's an example nginx.conf from one of my projects:

```
access_log $LOG_ROOT/$APP/access.log;
error_log $LOG_ROOT/$APP/error.log warn;
```

Then in ENV: `NGINX_INCLUDE_FILE=nginx.conf`

**Reply by jfmatth** [reply]
yea, brilliant idea, worked 100% for what i needed.

thanks again @chr15m  for the quick response.

**Reply by seguri** [reply]
Can you explain me this warning from `/home/piku/.piku/nginx/$APP.conf`?

```
# These are not required under systemd - enable for debugging only
# access_log        /home/piku/.piku/logs/$APP/access.log;
# error_log         /home/piku/.piku/logs/$APP/error.log;
```

They seem to contrast with your suggestion (that I would also apply)

**Reply by rcarmo** [reply]
When you're running things under `systemd`, some distros pipe the logs to a common location.

**Comment by chr15m** [comment]
I have no idea, sorry. I'm a fan of plain text logs I can tail, grep, or whatever so I try to keep it simple like that.

**Reply by jfmatth** [reply]
Yea, i'm starting to use Grafana's free tier and their NGINX integration is pretty good, but complicated, but your help made it easy.
